src/data_source/storage/sql/queries/node.rs

```diff
@@ -340,9 +341,18 @@ impl<Types: NodeType> UpdateAggregatesStorage<Types> for Transaction<Write> {
             "aggregate",
             ["height", "num_transactions", "payload_size"],
             ["height"],
-            rows,
+            rows.clone(),
```

Comment:
Nit: it is a bit wasteful to clone all the rows here just to grab the last row below. Might make more sense to pull out the updated aggregate first. Doesn't matter much though as the rows are usually small. This is not a blocker